packages/insomnia-app/app/ui/components/settings/import-export.tsx

```diff
@@ -1,98 +1,111 @@
-import React, { PureComponent } from 'react';
-import { autoBindMethodsForReact } from 'class-autobind-decorator';
-import { AUTOBIND_CFG } from '../../../common/constants';
+import React, { FC, useCallback } from 'react';
 import { Dropdown, DropdownButton, DropdownDivider, DropdownItem } from '../base/dropdown';
 import Link from '../base/link';
-import { showPrompt } from '../modals/index';
+import { showModal, showPrompt } from '../modals/index';
 import { docsImportExport } from '../../../common/documentation';
 import { strings } from '../../../common/strings';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectActiveSpaceName, selectActiveWorkspace } from '../../redux/selectors';
+import ExportRequestsModal from '../modals/export-requests-modal';
+import { exportAllToFile, importClipBoard, importFile, importUri } from '../../redux/modules/global';
+import { isDesign } from '../../../models/helpers/is-model';
 
 interface Props {
-  handleImportFile: Function;
-  handleImportClipBoard: Function;
-  handleImportUri: Function;
-  handleExportAll: Function;
-  handleShowExportRequestsModal: Function;
+  hideSettingsModal: () => void;
 }
 
-@autoBindMethodsForReact(AUTOBIND_CFG)
-class ImportExport extends PureComponent<Props> {
-  _handleImportUri() {
+export const ImportExport: FC<Props> = ({ hideSettingsModal }) => {
+  const dispatch = useDispatch();
+  const spaceName = useSelector(selectActiveSpaceName);
+  const activeWorkspace = useSelector(selectActiveWorkspace);
+  const activeResourceScope = isDesign(activeWorkspace) ? strings.document : strings.collection;
+
+  const handleImportUri = useCallback(() => {
     const lastUsedImportUri = window.localStorage.getItem('insomnia.lastUsedImportUri');
-    const promptOptions = {
+    const defaultValue = lastUsedImportUri ? { defaultValue: lastUsedImportUri } : {};
+    showPrompt({
       title: 'Import Data from URL',
       submitName: 'Fetch and Import',
       label: 'URL',
       placeholder: 'https://website.com/insomnia-import.json',
-      onComplete: uri => {
+      onComplete: async (uri: string) => {
         window.localStorage.setItem('insomnia.lastUsedImportUri', uri);
-        this.props.handleImportUri(uri);
+        await importUri(activeWorkspace._id, uri)(dispatch);
+        hideSettingsModal();
       },
-      ...(lastUsedImportUri ? { defaultValue: lastUsedImportUri } : {}),
-    };
+      ...defaultValue,
+    });
+  }, [hideSettingsModal, activeWorkspace, dispatch]);
+
+  const showExportRequestsModal = useCallback(() => {
+    showModal(ExportRequestsModal);
+    hideSettingsModal();
+  }, [hideSettingsModal]);
+
+  const handleExportAllToFile = useCallback(async () => {
+    dispatch(exportAllToFile);
```

Comment:
```suggestion
    await dispatch(exportAllToFile());
```

`exportAllToFile` needs to update too 🤔 from my understanding async actions themselves don't take dispatch, they return a function which accept dispatch (as well as state and any extra overrides, following redux-thunk docs)

```diff
- export const exportAllToFile = async (dispatch: Dispatch) => {
+ export const exportAllToFile = () => async (dispatch: Dispatch) => {
```

Reply:
this isn't relevant to redux-thunks because the invocation is evaluated before dispatch is called:

```ts

const callee = console.log;

const myFunction1 = () => async someArg => { console.log(someArg); };
callee(myFunction1());

const myFunction2 = async someArg => { console.log(someArg); };
callee(myFunction2);
```

these two should be equivalent.  So would be:

```ts
const myFunction3 = () => () => () => async someArg => { console.log(someArg); };
callee(myFunction3()()());
```

All redux-thunk sees is the result of the invocation.  I didn't see any reason to needlessly wrap, but if you have one I'm open minded.

There are others that suffer from the same problem that was fixed with exportAllToFile (e.g. `initActiveActivity`, `goToNextActivity` and two, `loadStart` and `loadStop` that are functions for no reason), but I opted to not touch the others because I don't want the changeset to expand here even more.  I did actually start fixing `initActiveActivity` for what it's worth but ended up backing out the changes once I saw that it would mean a lot more changes.  I can fix the ones that weren't immediately affected by this PR if you want in a separate PR?

Reply:
Ah! I see what you mean. Using the syntax in `myFunction1` is safer.

For example, if the action creator is updated to a function which receives a parameter, TS will complain correctly. If you don't invoke the action creator, TS will not complain.

As an example, lets say the action creator needs to now accept a parameter.

If invoking the action creator:
```diff
- const actionCreator = () => async dispatch => { ... };
+ const actionCreator = (arg: string) => async dispatch => { ... };

dispatch(actionCreator()); // TS will error at compile time
```

If not invoking the action creator like in your change:
```diff
- const actionCreator = async dispatch => { ... };
+ const actionCreator = (arg: string) => async dispatch => { ... };

dispatch(actionCreator); // TS will NOT error at compile time
```

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/4312346/121461442-f8550980-ca02-11eb-985d-8a5fed46cb94.png">

Reply:
I don't agree with you, but that's fine - I hear where you're coming from and now I understand much better why you wanted this change.  All good. :) updated in d210bd008221306c55a29e8671a51092693d7eda

Reply:
regarding
> I don't necessarily agree that the others are broken and so I am certainly opposed to changing them for the above reason

Take a look at the redux docs https://redux.js.org/tutorials/fundamentals/part-2-concepts-data-flow#actions

![Screenshot_20210610_061412](https://user-images.githubusercontent.com/15232461/121507923-14c95580-c9b3-11eb-8581-e23d1ec0acce.png)

and compare that to one of the actions I mentioned in the list of ones I felt were broken:

![Screenshot_20210610_061516](https://user-images.githubusercontent.com/15232461/121508067-3a565f00-c9b3-11eb-9b27-fa9496889d03.png)

I listed the specific ones instead of talking in generalities to try to make that part clear, but reading back I could have just shown this comparison to _actually_ make it clear.  Sorry bout that.